src/garage/tf/algos/rl2.py

```diff
@@ -1,15 +1,340 @@
 """Module for RL2.
 
-This module contains RL2Worker and the environment wrapper for RL2.
+This module contains RL2, RL2Worker and the environment wrapper for RL2.
 """
+import collections
+
 import akro
+from dowel import logger, tabular
 import gym
 import numpy as np
 
+import garage
+from garage import log_performance
+from garage import TrajectoryBatch
 from garage.envs import EnvSpec
+from garage.misc import tensor_utils as np_tensor_utils
+from garage.np.algos import MetaRLAlgorithm
 from garage.sampler import DefaultWorker
 
 
+class RL2(MetaRLAlgorithm):
+    """RL^2.
+
+    Reference: https://arxiv.org/pdf/1611.02779.pdf.
+
+    Args:
+        inner_algo (garage.np.algos.RLAlgorithm): Inner algorithm.
```

Comment:
it seems that the actual type of `garage.tf.algos.BatchPolopt`, based on your assert below.

Reply:
I'm pretty sure the actual type should be `garage.tf.algos.RL2NPO`.